core/src/main/java/jenkins/util/io/PathRemover.java

```diff
@@ -285,6 +289,15 @@ private static void makeWritable(@Nonnull Path path) throws IOException {
             } catch (UnsupportedOperationException ignored) {
                 // PosixFileAttributes not supported, fall back to old IO.
             }
+        } else {
+          /*
+           * If on Windows a folder has a read only attribute set, the file.setWritable(true) doesn't work (JENKINS-57855)
+           */
+          DosFileAttributeView dos = Files.getFileAttributeView(path, DosFileAttributeView.class, LinkOption.NOFOLLOW_LINKS);
+          if (dos != null)
+          {
+            dos.setReadOnly(false);
+          }
```

Comment:
Should it handle the SecurityException and wrap it into IOException? https://docs.oracle.com/javase/7/docs/api/java/nio/file/attribute/DosFileAttributeView.html#setReadOnly(boolean)

Reply:
Then we would need to wrap almost all File stuff in this class.

Reply:
fair point